cumulus_library/cli.py

```diff
@@ -194,166 +279,77 @@ def run_cli(args: Dict):
-        help=("Create a study instance from a template"),
-    )
-    studygen.add_argument(
-        "-p",
-        "--path",
-        default="./",
-        help=(
-            "The the directory the study will be created in. Default is "
-            "the current directory."
-        ),
-    )
-
-    db = parser.add_argument_group("SQL database modifications")
-    db.add_argument(
-        "-t",
-        "--target",
-        action="append",
-        help=(
-            "Specify one or more studies to create views form. Default is to "
-            "build all studies."
-        ),
-    )
-    db.add_argument(
-        "-s",
-        "--study-dir",
-        action="append",
-        help=(
-            "Optionally add one or more directories to look for study definitions in. "
-            "Default is in project directory and CUMULUS_LIBRARY_PATH, if present, "
-            "followed by any supplied paths. Target, and all its subdirectories, "
-            "are checked for manifests. Overriding studies with the same namespace "
-            "supersede earlier ones."
-        ),
-    )
-    db.add_argument(
-        "-b",
-        "--build",
-        default=False,
-        action="store_true",
-        help=("Removes and recreates Athena tables & views for the specified studies"),
-    )
-    db.add_argument(
-        "-e",
-        "--export",
-        default=False,
-        action="store_true",
-        help=("Generates files on disk from Athena views for the specified studies"),
-    )
-    db.add_argument(
-        "-v",
-        "--verbose",
-        default=False,
-        action="store_true",
-        help=("Prints detailed SQL query info"),
-    )
-
-    aws = parser.add_argument_group("AWS config")
-    aws.add_argument("--profile", help="AWS profile", default="default")
-    aws.add_argument(
-        "--database",
-        # internally, we use PyAthena's terminology for this but the UX term is "database"
-        dest="schema_name",
-        help="Cumulus Athena database name",
-    )
-    aws.add_argument(
-        "--workgroup",
-        default="cumulus",
-        help="Cumulus Athena workgroup (default: cumulus)",
-    )
-    aws.add_argument(
-        "--region",
-        help="AWS region data of Athena instance (default: us-east-1)",
-        default="us-east-1",
-    )
-
-    return parser
-
-
 def main(cli_args=None):
     """Reads CLI input/environment variables and invokes library calls"""
     parser = get_parser()
     args = vars(parser.parse_args(cli_args))
-
-    if args["target"] is not None:
+    if args["action"] is None:
+        parser.print_usage()
+        sys.exit(1)
+    if "target" in args and args["target"] is not None:
         for target in args["target"]:
             if target == "all":
                 args["target"] = ["all"]
                 break
-    else:
-        args["target"] = ["all"]
 
-    if args["study_dir"] is not None:
+    if "study_dir" in args and args["study_dir"] is not None:
```

Comment:
When is study_dir not in args? I thought Namespace would ensure the key exists at least. Unless you're dodging for tests?

Reply:
perhaps Namespace does, and because i'm casting it to a dict it doesn't? but study_dir is only in build and export, and maybe really should be renamed to be more 'additional paths to look for studies' somehow.

Reply:
Ah ok interesting that the `dict` version doesn't keep the empties.

OK, then I'm changing my comment to a code nit: `if args.get("study_dir"):` might be more readable.

Reply:
ok, touched all those dict accessors